test/endpoints/favourite_filters/conftest.py

```diff
@@ -0,0 +1,108 @@
+from bson import ObjectId
+from pymongo import MongoClient
+import pytest
+
+from operationsgateway_api.src.config import Config
+
+
+@pytest.fixture()
+def single_favourite_filter():
+    test_filter = FavouriteFilter()
+    inserted_filter = test_filter.create_one()
+
+    yield inserted_filter
+
+    test_filter.delete(inserted_filter["_id"])
+
+
+@pytest.fixture()
+def multiple_favourite_filters():
+    test_filter = FavouriteFilter()
+    test_filter.delete_all()
```

Comment:
Why do we need this in the setup of the fixture (before the `yield`)? If another test has generated a favourite filter that conflicts with this fixture then it should be the responsibility of that test to clear itself up (otherwise, if more tests are added and the order changes, or tests are selectively run then we can't guarantee that this `delete_all` will occur in the "right" place).

Reply:
This is a defensive measure to protect against filters in the database that shouldn't be - you can't guarantee someone hasn't been in the database and fiddled before running the tests! I've looked through the tests and I believe every test does clean up the filters it creates.